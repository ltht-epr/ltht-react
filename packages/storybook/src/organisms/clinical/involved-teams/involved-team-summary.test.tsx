import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import InvolvedTeamSummary from '@ltht-react/involved-team-summary'
import episodes from './involved-team-summary.fixtures'

describe('InvolvedTeamSummary', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<InvolvedTeamSummary episodeOfCares={episodes} />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<InvolvedTeamSummary episodeOfCares={episodes} />)).toMatchSnapshot('wrapper mount')
  })
})
