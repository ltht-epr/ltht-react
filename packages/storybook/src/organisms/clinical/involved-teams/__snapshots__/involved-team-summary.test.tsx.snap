// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvolvedTeamSummary matches snapshot: wrapper mount 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-4 > div {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
}

.emotion-0 {
  color: #768692;
}

.emotion-3 {
  text-align: right;
}

.emotion-2 {
  color: #231f20;
}

.emotion-5 {
  color: #231f20;
  text-align: left;
}

.emotion-6 {
  color: #768692;
  text-align: left;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-9 {
  text-align: right;
}

.emotion-8 {
  color: #768692;
  text-align: right;
  font-size: smaller;
  padding-top: 0.25rem;
}

<InvolvedTeamSummary
  episodeOfCares={
    Array [
      Object {
        "careManager": Object {
          "display": "Bella Swann (0113 111111)",
          "typeName": "Care Manager",
        },
        "id": "580ca927-34e0-e911-a2c7-005056926fe4|involvedteam",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": true,
          "requestedWhen": "",
        },
        "period": Object {
          "end": Object {
            "kind": "DATE",
            "value": "2020-01-02T00:00:00+00:00",
          },
          "start": Object {
            "kind": "DATE",
            "value": "2018-12-01T00:00:00+00:00",
          },
        },
        "status": "FINISHED",
        "team": Array [
          Object {
            "display": "LYPFT Outbound Care Team",
            "typeName": "Test Team",
          },
        ],
        "type": Array [
          Object {
            "coding": Array [
              Object {
                "code": "109007",
                "display": "Community Care",
              },
            ],
          },
        ],
      },
      Object {
        "careManager": Object {
          "display": "Joe Bloggs (0113 000000)",
          "typeName": "Care Manager",
        },
        "id": "690ca927-68e0-c511-b2c7-898056926ze4|involvedteam",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": false,
          "requestedWhen": "",
        },
        "period": Object {
          "start": Object {
            "kind": "DATE",
            "value": "2019-12-01T00:00:00+00:00",
          },
        },
        "status": "ACTIVE",
        "team": Array [
          Object {
            "display": "LYPFT Inbound Care Team",
            "typeName": "Test Team",
          },
        ],
        "type": Array [
          Object {
            "coding": Array [
              Object {
                "code": "109016",
                "display": "Home Care",
              },
            ],
          },
        ],
      },
    ]
  }
>
  <ListItem
    key="580ca927-34e0-e911-a2c7-005056926fe4|involvedteam"
  >
    <li>
      <InvolvedTeamSummaryItem
        episodeOfCare={
          Object {
            "careManager": Object {
              "display": "Bella Swann (0113 111111)",
              "typeName": "Care Manager",
            },
            "id": "580ca927-34e0-e911-a2c7-005056926fe4|involvedteam",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": true,
              "requestedWhen": "",
            },
            "period": Object {
              "end": Object {
                "kind": "DATE",
                "value": "2020-01-02T00:00:00+00:00",
              },
              "start": Object {
                "kind": "DATE",
                "value": "2018-12-01T00:00:00+00:00",
              },
            },
            "status": "FINISHED",
            "team": Array [
              Object {
                "display": "LYPFT Outbound Care Team",
                "typeName": "Test Team",
              },
            ],
            "type": Array [
              Object {
                "coding": Array [
                  Object {
                    "code": "109007",
                    "display": "Community Care",
                  },
                ],
              },
            ],
          }
        }
      >
        <div
          className="emotion-4"
          role="link"
        >
          <InvolvedTeamRedacted
            episodeOfCare={
              Object {
                "careManager": Object {
                  "display": "Bella Swann (0113 111111)",
                  "typeName": "Care Manager",
                },
                "id": "580ca927-34e0-e911-a2c7-005056926fe4|involvedteam",
                "metadata": Object {
                  "dataSources": Array [
                    Object {
                      "display": "Mock",
                    },
                  ],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "period": Object {
                  "end": Object {
                    "kind": "DATE",
                    "value": "2020-01-02T00:00:00+00:00",
                  },
                  "start": Object {
                    "kind": "DATE",
                    "value": "2018-12-01T00:00:00+00:00",
                  },
                },
                "status": "FINISHED",
                "team": Array [
                  Object {
                    "display": "LYPFT Outbound Care Team",
                    "typeName": "Test Team",
                  },
                ],
                "type": Array [
                  Object {
                    "coding": Array [
                      Object {
                        "code": "109007",
                        "display": "Community Care",
                      },
                    ],
                  },
                ],
              }
            }
          >
            <div
              className="emotion-1"
            >
              <RedactedDescription>
                <div
                  className="emotion-0"
                >
                  Insufficient privileges
                </div>
              </RedactedDescription>
            </div>
            <div
              className="emotion-3"
            >
              <PeriodSummary
                period={
                  Object {
                    "end": Object {
                      "kind": "DATE",
                      "value": "2020-01-02T00:00:00+00:00",
                    },
                    "start": Object {
                      "kind": "DATE",
                      "value": "2018-12-01T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Dec-01,-2018 to Jan-02,-2020
                </div>
              </PeriodSummary>
            </div>
          </InvolvedTeamRedacted>
        </div>
      </InvolvedTeamSummaryItem>
    </li>
  </ListItem>
  <ListItem
    key="690ca927-68e0-c511-b2c7-898056926ze4|involvedteam"
  >
    <li>
      <InvolvedTeamSummaryItem
        episodeOfCare={
          Object {
            "careManager": Object {
              "display": "Joe Bloggs (0113 000000)",
              "typeName": "Care Manager",
            },
            "id": "690ca927-68e0-c511-b2c7-898056926ze4|involvedteam",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": false,
              "requestedWhen": "",
            },
            "period": Object {
              "start": Object {
                "kind": "DATE",
                "value": "2019-12-01T00:00:00+00:00",
              },
            },
            "status": "ACTIVE",
            "team": Array [
              Object {
                "display": "LYPFT Inbound Care Team",
                "typeName": "Test Team",
              },
            ],
            "type": Array [
              Object {
                "coding": Array [
                  Object {
                    "code": "109016",
                    "display": "Home Care",
                  },
                ],
              },
            ],
          }
        }
      >
        <div
          className="emotion-4"
          role="link"
        >
          <div>
            <InvolvedTeamTitle
              episodeOfCare={
                Object {
                  "careManager": Object {
                    "display": "Joe Bloggs (0113 000000)",
                    "typeName": "Care Manager",
                  },
                  "id": "690ca927-68e0-c511-b2c7-898056926ze4|involvedteam",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "period": Object {
                    "start": Object {
                      "kind": "DATE",
                      "value": "2019-12-01T00:00:00+00:00",
                    },
                  },
                  "status": "ACTIVE",
                  "team": Array [
                    Object {
                      "display": "LYPFT Inbound Care Team",
                      "typeName": "Test Team",
                    },
                  ],
                  "type": Array [
                    Object {
                      "coding": Array [
                        Object {
                          "code": "109016",
                          "display": "Home Care",
                        },
                      ],
                    },
                  ],
                }
              }
            >
              <div
                className="emotion-5"
              >
                LYPFT Inbound Care Team
              </div>
            </InvolvedTeamTitle>
            <InvolvedTeamDescription
              episodeOfCare={
                Object {
                  "careManager": Object {
                    "display": "Joe Bloggs (0113 000000)",
                    "typeName": "Care Manager",
                  },
                  "id": "690ca927-68e0-c511-b2c7-898056926ze4|involvedteam",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "period": Object {
                    "start": Object {
                      "kind": "DATE",
                      "value": "2019-12-01T00:00:00+00:00",
                    },
                  },
                  "status": "ACTIVE",
                  "team": Array [
                    Object {
                      "display": "LYPFT Inbound Care Team",
                      "typeName": "Test Team",
                    },
                  ],
                  "type": Array [
                    Object {
                      "coding": Array [
                        Object {
                          "code": "109016",
                          "display": "Home Care",
                        },
                      ],
                    },
                  ],
                }
              }
            >
              <div
                className="emotion-6"
              >
                Joe Bloggs (0113 000000)
              </div>
            </InvolvedTeamDescription>
          </div>
          <div
            className="emotion-9"
          >
            <PeriodSummary
              period={
                Object {
                  "start": Object {
                    "kind": "DATE",
                    "value": "2019-12-01T00:00:00+00:00",
                  },
                }
              }
            >
              <div
                className="emotion-2"
              >
                Dec-01,-2019 to Present
              </div>
            </PeriodSummary>
            <InvolvedTeamType
              episodeOfCare={
                Object {
                  "careManager": Object {
                    "display": "Joe Bloggs (0113 000000)",
                    "typeName": "Care Manager",
                  },
                  "id": "690ca927-68e0-c511-b2c7-898056926ze4|involvedteam",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "period": Object {
                    "start": Object {
                      "kind": "DATE",
                      "value": "2019-12-01T00:00:00+00:00",
                    },
                  },
                  "status": "ACTIVE",
                  "team": Array [
                    Object {
                      "display": "LYPFT Inbound Care Team",
                      "typeName": "Test Team",
                    },
                  ],
                  "type": Array [
                    Object {
                      "coding": Array [
                        Object {
                          "code": "109016",
                          "display": "Home Care",
                        },
                      ],
                    },
                  ],
                }
              }
            >
              <div
                className="emotion-8"
              >
                Active
              </div>
            </InvolvedTeamType>
          </div>
        </div>
      </InvolvedTeamSummaryItem>
    </li>
  </ListItem>
</InvolvedTeamSummary>
`;
